ingestion-beam/src/main/java/com/mozilla/telemetry/util/DynamicPathTemplate.java

```diff
@@ -142,7 +142,7 @@ public String replaceDynamicPart(List<String> values) {
       // This regex matches either of these patterns,
       // capturing the name of the placeholder as group 1
       // and the default value (or null if no default is provided) as group 2.
-      "\\$\\{(\\p{Alnum}+)(?::-(\\p{Alnum}+))?}"
+      "\\$\\{([^{}]+?(?=:-|}))(?::-([^{}]+))?}"
```

Comment:
Updated the regex and documented the restrictions on naming.

The new regex allows all characters other than curly braces and relies on a reluctant quantifier (`+?`) and positive lookahead (`(?=:-|})`) to determine the end of the attribute name.